core-metastore/src/main/java/org/zalando/nakadi/repository/db/SubscriptionDbRepository.java

```diff
@@ -142,16 +226,43 @@ public void deleteSubscription(final String id)
                     .map(et -> format("\"{0}\"", et))
                     .forEach(params::add);
         }
+
+        final String order;
+        if (token == null) {
+            order = " ORDER BY s_subscription_object->>'created_at' DESC LIMIT ? OFFSET ? ";
+            params.add(limit);
+            params.add(offset);
+        } else {
+            if (!token.subscriptionId.isEmpty()) {
+                if (token.getTokenType() == TokenType.FORWARD) {
+                    clauses.add(" s_id > ? ");
+                } else if (token.getTokenType() == TokenType.BACKWARD){
+                    clauses.add(" s_id < ? ");
+                } else if (token.getTokenType() == TokenType.BACKWARD_INCL) {
+                    clauses.add(" s_id <= ?");
+                } else {
+                    throw new IllegalStateException("Token type is not supported");
+                }
+                params.add(token.subscriptionId);
+            }
+            if (token.getTokenType() == TokenType.FORWARD) {
+                order = " ORDER BY s_id LIMIT ?";
+            } else {
+                order = " ORDER BY s_id DESC LIMIT ?";
+            }
+            params.add(limit);
+        }
         if (!clauses.isEmpty()) {
             queryBuilder.append(" WHERE ");
             queryBuilder.append(StringUtils.join(clauses, " AND "));
         }
-
-        queryBuilder.append(" ORDER BY s_subscription_object->>'created_at' DESC LIMIT ? OFFSET ? ");
-        params.add(limit);
-        params.add(offset);
+        queryBuilder.append(order);
         try {
-            return jdbcTemplate.query(queryBuilder.toString(), params.toArray(), rowMapper);
+            final List<Subscription> result = jdbcTemplate.query(queryBuilder.toString(), params.toArray(), rowMapper);
+            if (null != token && token.getTokenType() != TokenType.FORWARD) {
+                Collections.reverse(result);
```

Comment:
Why reverse when there is ` ORDER BY s_id DESC LIMIT ?`

Reply:
I rewrote code in a way that it is specifically written why